modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/ags_solver.cc

```diff
@@ -36,13 +36,45 @@ AGSSolver::Solve()
       solver->Solve();
     }
 
-    double pw_change = ComputePointwisePhiChange();
-    double rho = (iter == 0) ? 0.0 : sqrt(pw_change / pw_change_prev);
-    pw_change_prev = pw_change;
+    if (lbs_solver_.Options().ags_pointwise_convergence)
+    {
+      double pw_change = ComputePointwisePhiChange();
+      double rho = (iter == 0) ? 0.0 : sqrt(pw_change / pw_change_prev);
+      pw_change_prev = pw_change;
+
+      if (pw_change < std::max(tolerance_ * (1.0 - rho), 1.0e-10))
+        converged = true;
 
-    if (pw_change < std::max(tolerance_ * (1.0 - rho), 1.0e-10))
-      converged = true;
+      if (verbose_)
+      {
+        std::stringstream iter_stats;
+        iter_stats << program_timer.GetTimeString() << " AGS Iteration " << std::left
+                   << std::setw(5) << iter << " Point-wise change " << std::left << std::setw(14)
+                   << pw_change << " Spectral-radius estimate " << std::left << std::setw(10)
+                   << rho;
+        if (converged)
+          iter_stats << " CONVERGED";
+        log.Log() << iter_stats.str();
+      }
```

Comment:
Consider unifying the printers and using a check to write point-wise or l2 instead of duplicating the whole block.

Reply:
I'll see if I can clean this up a bit.

Reply:
Done.